Added parameters on RequestModels for using new OpenAI features

OmarSilviniDocsMarshal opened the pull request:
Added the Attributes parameter to the CreateVectorStoreFileRequest to "decorate" the files in order to use the attribute filtering of the file search tool.
Added the Filters parameter to the FileSearchTool that allows to filter the files that the assistant will use for searching in vector db
Added the ReasoningEffort parameter to the RunCreateRequest that allows to use the newest reasoning models

OmarSilviniDocsMarshal commented:
@dotnet-policy-service agree


[cid:f49a81cf-921f-4545-b56f-3222fc3f563e]<https://www.cualeva.com/>



________________________________
From: dotnet-policy-service[bot] ***@***.***>
Sent: Thursday, June 5, 2025 11:53 AM
To: betalgo/openai ***@***.***>
Cc: Omar Silvini ***@***.***>; Mention ***@***.***>
Subject: Re: [betalgo/openai] Added parameters on RequestModels for using new OpenAI features (PR #721)

[https://avatars.githubusercontent.com/in/285228?s=20&v=4]dotnet-policy-service[bot] left a comment (betalgo/openai#721)<https://github.com/betalgo/openai/pull/721#issuecomment-2943517965>

@OmarSilviniDocsMarshal<https://github.com/OmarSilviniDocsMarshal> please read the following Contributor License Agreement(CLA). If you agree with the CLA, please reply with the following information.

@dotnet-policy-service agree [company="{your company}"]


Options:

  *   (default - no company specified) I have sole ownership of intellectual property rights to my Submissions and I am not making Submissions in the course of work for my employer.

@dotnet-policy-service agree


  *   (when company given) I am making Submissions in the course of work for my employer (or my employer has intellectual property rights in my Submissions by contract or applicable law). I have permission from my employer to make Submissions and enter into this Agreement on behalf of my employer. By signing below, the defined term “You” includes me and my employer.

@dotnet-policy-service agree company="Microsoft"


Contributor License Agreement
Contribution License Agreement

This Contribution License Agreement ( “Agreement” ) is agreed to by the party signing below ( “You” ),
and conveys certain license rights to the .NET Foundation ( “.NET Foundation” ) for Your contributions to
.NET Foundation open source projects. This Agreement is effective as of the latest signature date below.

1. Definitions.

“Code” means the computer software code, whether in human-readable or machine-executable form,
that is delivered by You to .NET Foundation under this Agreement.

“Project” means any of the projects owned or managed by .NET Foundation and offered under a license
approved by the Open Source Initiative (www.opensource.org<http://www.opensource.org>).

“Submit” is the act of uploading, submitting, transmitting, or distributing code or other content to any
Project, including but not limited to communication on electronic mailing lists, source code control
systems, and issue tracking systems that are managed by, or on behalf of, the Project for the purpose of
discussing and improving that Project, but excluding communication that is conspicuously marked or
otherwise designated in writing by You as “Not a Submission.”

“Submission” means the Code and any other copyrightable material Submitted by You, including any
associated comments and documentation.

2. Your Submission. You must agree to the terms of this Agreement before making a Submission to any
Project. This Agreement covers any and all Submissions that You, now or in the future (except as
described in Section 4 below), Submit to any Project.

3. Originality of Work. You represent that each of Your Submissions is entirely Your
original work. Should You wish to Submit materials that are not Your original work,
You may Submit them separately to the Project if You (a) retain all copyright and
license information that was in the materials as you received them, (b) in the
description accompanying your Submission, include the phrase "Submission
containing materials of a third party:" followed by the names of the third party and any
licenses or other restrictions of which You are aware, and (c) follow any other
instructions in the Project's written guidelines concerning Submissions.

4. Your Employer. References to “employer” in this Agreement include Your employer or anyone else
for whom You are acting in making Your Submission, e.g. as a contractor, vendor, or agent. If Your
Submission is made in the course of Your work for an employer or Your employer has intellectual
property rights in Your Submission by contract or applicable law, You must secure permission from Your
employer to make the Submission before signing this Agreement. In that case, the term “You” in this
Agreement will refer to You and the employer collectively. If You change employers in the future and
desire to Submit additional Submissions for the new employer, then You agree to sign a new Agreement
and secure permission from the new employer before Submitting those Submissions.

5. Licenses.

a. Copyright License. You grant .NET Foundation, and those who receive the Submission directly
or indirectly from .NET Foundation, a perpetual, worldwide, non-exclusive, royalty-free, irrevocable
license in the Submission to reproduce, prepare derivative works of, publicly display, publicly perform,
and distribute the Submission and such derivative works, and to sublicense any or all of the foregoing
rights to third parties.

b. Patent License. You grant .NET Foundation, and those who receive the Submission directly or
indirectly from .NET Foundation, a perpetual, worldwide, non-exclusive, royalty-free, irrevocable license
under Your patent claims that are necessarily infringed by the Submission or the combination of the
Submission with the Project to which it was Submitted to make, have made, use, offer to sell, sell and
import or otherwise dispose of the Submission alone or with the Project.

c. Other Rights Reserved. Each party reserves all rights not expressly granted in this Agreement.
No additional licenses or rights whatsoever (including, without limitation, any implied licenses) are
granted by implication, exhaustion, estoppel or otherwise.

6. Representations and Warranties. You represent that You are legally entitled to grant the above
licenses. You represent that each of Your Submissions is entirely Your original work (except as You may
have disclosed under Section 3 ). You represent that You have secured permission from Your employer to
make the Submission in cases where Your Submission is made in the course of Your work for Your
employer or Your employer has intellectual property rights in Your Submission by contract or applicable
law. If You are signing this Agreement on behalf of Your employer, You represent and warrant that You
have the necessary authority to bind the listed employer to the obligations contained in this Agreement.
You are not expected to provide support for Your Submission, unless You choose to do so. UNLESS
REQUIRED BY APPLICABLE LAW OR AGREED TO IN WRITING, AND EXCEPT FOR THE WARRANTIES
EXPRESSLY STATED IN SECTIONS 3, 4, AND 6 , THE SUBMISSION PROVIDED UNDER THIS AGREEMENT IS
PROVIDED WITHOUT WARRANTY OF ANY KIND, INCLUDING, BUT NOT LIMITED TO, ANY WARRANTY OF
NONINFRINGEMENT, MERCHANTABILITY, OR FITNESS FOR A PARTICULAR PURPOSE.

7. Notice to .NET Foundation. You agree to notify .NET Foundation in writing of any facts or
circumstances of which You later become aware that would make Your representations in this
Agreement inaccurate in any respect.

8. Information about Submissions. You agree that contributions to Projects and information about
contributions may be maintained indefinitely and disclosed publicly, including Your name and other
information that You submit with Your Submission.

9. Governing Law/Jurisdiction. This Agreement is governed by the laws of the State of Washington, and
the parties consent to exclusive jurisdiction and venue in the federal courts sitting in King County,
Washington, unless no federal subject matter jurisdiction exists, in which case the parties consent to
exclusive jurisdiction and venue in the Superior Court of King County, Washington. The parties waive all
defenses of lack of personal jurisdiction and forum non-conveniens.

10. Entire Agreement/Assignment. This Agreement is the entire agreement between the parties, and
supersedes any and all prior agreements, understandings or communications, written or oral, between
the parties relating to the subject matter hereof. This Agreement may be assigned by .NET Foundation.

.NET Foundation dedicates this Contribution License Agreement to the public domain according to the Creative Commons CC0 1.

—
Reply to this email directly, view it on GitHub<https://github.com/betalgo/openai/pull/721#issuecomment-2943517965>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/BI3FVHARBJWXP65IZKQCOTD3CAHSXAVCNFSM6AAAAAB6U6LD72VHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDSNBTGUYTOOJWGU>.
You are receiving this because you were mentioned.Message ID: ***@***.***>

<https://www.certipedia.com/quality_marks/9000034730>
[Certificato ISO Cualeva srl]

kayhantolga commented:
Hi @OmarSilviniDocsMarshal, could you please share the link to the source documents for these changes so I can verify them?

OmarSilviniDocsMarshal commented:
> source documents

Hi, I changed some models model following the API documentation linked below:
- RunCreateRequest -  [CreateRun](https://platform.openai.com/docs/api-reference/runs/createRun)
- CreateVectorStoreFileRequest - [CreateVectorStoreFile](https://platform.openai.com/docs/api-reference/vector-stores-files/createFile)
- FileSearchTool - [AttributeFiltering](https://platform.openai.com/docs/guides/retrieval?attributes-filter-example=region#attribute-filtering)

I also added the summary to the properties...
